src/main/java/pl/agawesolowska/ticketbookingapp/facade/BookingFacade.java

```diff
@@ -0,0 +1,81 @@
+package pl.agawesolowska.ticketbookingapp.facade;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import pl.agawesolowska.ticketbookingapp.model.dto.BookingRequestDTO;
+import pl.agawesolowska.ticketbookingapp.model.dto.BookingResultDTO;
+import pl.agawesolowska.ticketbookingapp.model.entity.Booking;
+import pl.agawesolowska.ticketbookingapp.model.entity.Customer;
+import pl.agawesolowska.ticketbookingapp.model.entity.Screening;
+import pl.agawesolowska.ticketbookingapp.model.entity.Seat;
+import pl.agawesolowska.ticketbookingapp.service.BookingService;
+import pl.agawesolowska.ticketbookingapp.service.CustomerService;
+import pl.agawesolowska.ticketbookingapp.service.ScreeningService;
+import pl.agawesolowska.ticketbookingapp.service.SeatService;
+import pl.agawesolowska.ticketbookingapp.util.DateTimeUtils;
+import pl.agawesolowska.ticketbookingapp.util.SeatReservationUtils;
+
+/**
+ * @author Aga Wesołowska
+ *
+ */
+@Component
+public class BookingFacade {
+
+	private final BookingService bookingService;
+	private final CustomerService customerService;
+	private final ScreeningService screeningService;
+	private final SeatService seatService;
+
+	@Autowired
+	public BookingFacade(BookingService bookingService, CustomerService customerService,
+			ScreeningService screeningService, SeatService seatService) {
+		this.bookingService = bookingService;
+		this.customerService = customerService;
+		this.screeningService = screeningService;
+		this.seatService = seatService;
+	}
+
+	public BookingResultDTO addCustomerBooking(BookingRequestDTO bookingRequestDTO) {
```

Comment:
potrzebujemy suffixu DTO?

Reply:
I tak, i nie. Wiedząc, że DTO to Data Transfer Object, czyli klatka schodowa którą wchodzą requesty i wychodzą resultsy, nie ma takiej konieczności. Ale wolę to dodać dla czytelności/przejrzystości kodu. Podobnie jak w przypadku adnotacji `@Autowired` - w wyższych wersjach Springa nie jest ona niezbędna, jeśli występuje tylko jeden konstruktor, ale uważam, że warto ją dodać dla jasności (szczególnie mi na początku, gdy analizuję i próbuję zrozumieć kod).